x-pack/test/plugin_api_integration/test_suites/task_manager/task_management.ts

```diff
@@ -646,77 +646,67 @@ export default function ({ getService }: FtrProviderContext) {
+        const task = await currentTask(scheduledTask.id);
 
-        expect(getTaskById(tasks, scheduledTask.id).enabled).to.eql(true);
+        expect(task.enabled).to.eql(true);
       });
 
       // disable the task
       await bulkDisable([scheduledTask.id]);
 
       await retry.try(async () => {
-        const tasks = (await currentTasks()).docs;
-
-        expect(getTaskById(tasks, scheduledTask.id).enabled).to.eql(false);
+        const task = await currentTask(scheduledTask.id);
+        expect(task.enabled).to.eql(false);
       });
 
       // re-enable the task
       await bulkEnable([scheduledTask.id], true);
 
       await retry.try(async () => {
-        const tasks = (await currentTasks()).docs;
+        const task = await currentTask(scheduledTask.id);
 
-        expect(getTaskById(tasks, scheduledTask.id).enabled).to.eql(true);
-
-        // should get a new document even tho original schedule interval was 30m
-        expect((await historyDocs()).length).to.eql(2);
+        expect(task.enabled).to.eql(true);
+        expect(Date.parse(task.scheduledAt)).to.greaterThan(Date.parse(scheduledTask.scheduledAt));
       });
     });
 
     it('should disable and reenable task and not run it when runSoon = false', async () => {
       const historyItem = random(1, 100);
       const scheduledTask = await scheduleTask({
         taskType: 'sampleTask',
-        schedule: { interval: '30m' },
+        schedule: { interval: '1h' },
         params: { historyItem },
       });
 
       await retry.try(async () => {
         expect((await historyDocs()).length).to.eql(1);
-        const tasks = (await currentTasks()).docs;
 
-        expect(getTaskById(tasks, scheduledTask.id).enabled).to.eql(true);
+        const task = await currentTask(scheduledTask.id);
+        expect(task.enabled).to.eql(true);
       });
 
       // disable the task
       await bulkDisable([scheduledTask.id]);
 
       await retry.try(async () => {
-        const tasks = (await currentTasks()).docs;
-
-        expect(getTaskById(tasks, scheduledTask.id).enabled).to.eql(false);
+        const task = await currentTask(scheduledTask.id);
+        expect(task.enabled).to.eql(false);
       });
 
       // re-enable the task
       await bulkEnable([scheduledTask.id], false);
 
       await retry.try(async () => {
-        const tasks = (await currentTasks()).docs;
-
-        const task = getTaskById(tasks, scheduledTask.id);
+        const task = await currentTask(scheduledTask.id);
 
         expect(task.enabled).to.eql(true);
-
-        // task runAt should be set in the future by greater than 20 minutes
-        // this assumes it takes less than 10 minutes to disable and renable the task
-        // since the schedule interval is 30 minutes
-        expect(Date.parse(task.runAt) - Date.now()).to.be.greaterThan(10 * 60 * 1000);
+        expect(Date.parse(task.scheduledAt)).to.eql(Date.parse(scheduledTask.scheduledAt));
```

Comment:
enabling should also not reset the `runAt` time, can we add an expect check for that as well?

Reply:
The runAt times are slightly different so that check fails. I am not sure why they are different.
``` 
task.runAt: "2022-09-26T13:34:45.768Z"
task.scheduledAt: "2022-09-26T12:34:42.516Z"
scheduledTask.runAt: "2022-09-26T12:34:42.516Z"
scheduledTask.scheduledAt: "2022-09-26T12:34:42.516Z"
```

Reply:
Oh I guess it is because it ran once so the `runAt` got reset from the initial schedule. I think the runAt for this should match the task from line 701, when it got disabled.

Reply:
Resolved in https://github.com/elastic/kibana/pull/141441/commits/ccb9e978161355ef33e08882cdf9a812114cc041